perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py

```diff
@@ -238,16 +243,63 @@ def _delete_existing_data_in_gcs_bucket(gcs_bucket)->(int):
         'gcloud alpha storage rm -r gs://{}/*'.format(gcs_bucket), shell=True)
     return 0
   except subprocess.CalledProcessError as e:
-    _logmessage(e.output.decode('utf-8'))
+    _logmessage(e.output.decode('utf-8'),"error")
     return 1
 
 
+def _generate_files_and_upload_to_gcs_bucket(destination_blob_name, num_of_files,
+    file_size_unit, file_size,
+    filename_prefix) -> int:
+  for batch_start in range(1, num_of_files + 1, BATCH_SIZE):
+    for file_num in range(batch_start, batch_start + BATCH_SIZE):
+      if file_num > num_of_files:
+        break
+
+      file_name = '{}_{}'.format(filename_prefix, file_num)
+      temp_file = '{}/{}.txt'.format(TEMPORARY_DIRECTORY, file_name)
+
+      # Creating files in temporary folder:
+      with open(temp_file, 'wb') as out:
+        if (file_size_unit.lower() == 'gb'):
+          out.truncate(1024 * 1024 * 1024 * int(file_size))
+        if (file_size_unit.lower() == 'mb'):
+          out.truncate(1024 * 1024 * int(file_size))
+        if (file_size_unit.lower() == 'kb'):
+          out.truncate(1024 * int(file_size))
+        if (file_size_unit.lower() == 'b'):
+          out.truncate(int(file_size))
+
+    num_files = os.listdir(TEMPORARY_DIRECTORY)
+
+    if not num_files:
```

Comment:
Compare num_files with actual number

Reply:
Will get complicated when number of files to be created is not a multiple of 100.